core/src/com/unciv/logic/automation/unit/BattleHelper.kt

```diff
@@ -121,8 +121,8 @@ object BattleHelper {
         if (tileCombatant is CityCombatant && tileCombatant.city.hasJustBeenConquered) return false
         if (!combatant.getCivInfo().isAtWarWith(tileCombatant.getCivInfo())) return false
 
-        if (combatant is MapUnitCombatant && combatant.isLandUnit() && combatant.isMelee() &&
-            !combatant.hasUnique(UniqueType.LandUnitEmbarkation) && tile.isWater
+        if (combatant is MapUnitCombatant && combatant.isLandUnit() && combatant.isMelee() && tile.isWater &&
+            (!combatant.hasUnique(UniqueType.LandUnitEmbarkation) && !combatant.hasUnique(UniqueType.CanMoveOnWater))
         )
```

Comment:
Function uses ICombatant, not MapUnit, seemingly to unify the city and unit attack code. The side effect is we can't access the cached values

Reply:
Actually, misunderstood your question... That's... Weird. I assume the intent is somewhere they thought that maybe there was a way to let only 1 unit embark? Idk. Looks like a big to me

Reply:
Looking through it, the tutorial for embarking almost makes sense. If this was a cache foor units, then for the tutorial to see that is should explain embarking, it would need to dig through every unit. Though... It's a cache on techs for some reason, so even that's questionable

Looking at this case for now, this strictly looks like a bug. The intent I think should be to allow attacking into watrer if the unit can embark. There's basically no reason to put LandUnitEmbarkation on a unit itself, so this should reference the cache...

Except, imo this actually should just be a cache of the unit, not the tech. I guess the's memory savings to put this on the tech manger, but reading the hoops `TechManger.specificUnitsCanEnterOcean` goes through, this just makes more sense on the units themselves. I won't fix that in this PR, just instead referencing the tech cache, but this should just be moved around